src/Routing/Middleware/SubstituteBindings.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+namespace Spatie\Enum\Laravel\Routing\Middleware;
+
+use Closure;
+use Illuminate\Contracts\Routing\Registrar;
+use Spatie\Enum\Laravel\Routing\ImplicitRouteBinding;
+
+class SubstituteBindings
+{
+    /**
+     * The router instance.
+     *
+     * @var \Illuminate\Contracts\Routing\Registrar
+     */
+    protected $router;
+
+    public function __construct(Registrar $router)
+    {
+        $this->router = $router;
+    }
+
+    /**
+     * @param \Illuminate\Http\Request $request
+     * @param \Closure $next
+     * @return mixed
+     */
+    public function handle($request, Closure $next)
+    {
+        $this->router->enumSubstituteImplicitBindings($request->route());
```

Comment:
If we don't need a dedicated class, we can just inline the resolve logic inside middleware.

Following is the code I wrote for my project:

```php
final class ImplicitEnumRouteBinding
{
    public function __construct(
        private Registrar $router,
        private Container $container
    )
    {
    }

    public function handle(Request $request, Closure $next)
    {
        $route = $request->route();
        $params = $route->parameters();

        foreach ($route->signatureParameters(Enum::class) as $param) {
            $paramName = self::getParamName($param->getName(), $params);

            $value = $params[$paramName];

            if ($value instanceof Enum) {
                continue;
            }


            $route->setParameter(
                $paramName,
                $this->container->make(Reflector::getParameterClassName($param), ['value' => $value])
            );
        }

        return $next($request);
    }

    private static function getParamName(string $name, array $params): ?string
    {
        if (array_key_exists($name, $params)) {
            return $name;
        }

        if (array_key_exists($snaked = Str::snake($name), $params)) {
            return $snaked;
        }

        return null;
    }
}
```